packages/blockchain-utils-linking/src/near.ts

```diff
@@ -0,0 +1,51 @@
+import { AccountID } from 'caip';
+import { AuthProvider } from './auth-provider';
+import { getConsentMessage, LinkProof } from './util';
+import { hash } from '@stablelib/sha256';
+import * as uint8arrays from 'uint8arrays';
+
+const stringEncode = (str: string): string => uint8arrays.toString(uint8arrays.fromString(str), 'base64pad');
+
+// REF: near.org
+export class NearAuthProvider implements AuthProvider {
+  readonly isAuthProvider = true;
+
+  constructor(
+    private readonly provider: any,
+    // NOTE: address here is: ed25519 public key
+    private readonly address: string,
+    private readonly chainRef: string) {}
+
+  async authenticate(message: string): Promise<string> {
```

Comment:
Can you add a test for this method to show that it always returns the same entropy bytes for the same near address?

Reply:
yes, will do